jerry-core/ecma/builtin-objects/ecma-builtin-regexp.c

```diff
@@ -112,7 +112,16 @@ ecma_builtin_regexp_dispatch_construct (const ecma_value_t *arguments_list_p, /*
 
     if (ecma_is_value_empty (ret_value))
     {
-      ret_value = ecma_op_create_regexp_object (pattern_string_p, flags_string_p);
+      uint16_t flags = 0;
+      if (ecma_is_value_empty (ret_value) && (flags_string_p != NULL))
```

Comment:
Isn't this a bit overly complex? `ecma_is_value_empty (ret_value)` is now tested twice (in the outer `if` and inside as well). The outer `if` is not needed IMHO.